0011-R-multiple_reminders_actors_impl.md

```diff
@@ -0,0 +1,149 @@
+
+## Implementation Details
+
+### Design
+
+Reminders / Actors should have their Interfaces segregated in such a way that it shuld be possible to implement them in an independent way.
+There are two places where APIs or functions required for Actors and Reminders are defined.
+
+1. Protos in dapr/proto package.
+
+2. Interfaces in pkg/actors/internal for placement and reminder.
+
+Out of these 2, protos are not to be updated, as every implementation will have their own RPCs defined accordingly.
+
+Only interfaces in pkg/actors/internal have to be updated. A major work has already been done for them in release 1.12 already by segregating actors, reminders and timers interfaces (https://github.com/dapr/dapr/pull/6669).
+
+```go
+
+// PlacementService allows for interacting with the actor placement service.
+type PlacementService interface {
+	io.Closer
+
+	Start(context.Context) error
+	WaitUntilReady(ctx context.Context) error
+	LookupActor(ctx context.Context, req LookupActorRequest) (LookupActorResponse, error)
+	AddHostedActorType(actorType string, idleTimeout time.Duration) error
+	ReportActorDeactivation(ctx context.Context, actorType, actorID string) error
+
+	SetHaltActorFns(haltFn HaltActorFn, haltAllFn HaltAllActorsFn)
+	SetOnAPILevelUpdate(fn func(apiLevel uint32))
+
+	// PlacementHealthy returns true if the placement service is healthy.
+	PlacementHealthy() bool
+	// StatusMessage returns a custom status message.
+	StatusMessage() string
+}
+
+```
+
+```go
+type RemindersProvider interface {
+	io.Closer
+
+	Init(ctx context.Context) error
+	GetReminder(ctx context.Context, req *GetReminderRequest) (*Reminder, error)
+	CreateReminder(ctx context.Context, req *Reminder) error
+	DeleteReminder(ctx context.Context, req DeleteReminderRequest) error
+	DrainRebalancedReminders(actorType string, actorID string)
+	OnPlacementTablesUpdated(ctx context.Context)
+
+	SetResiliencyProvider(resiliency resiliency.Provider)
+	SetExecuteReminderFn(fn ExecuteReminderFn)
+	SetStateStoreProviderFn(fn StateStoreProviderFn)
+	SetLookupActorFn(fn LookupActorFn)
+}
+
+```
+Idea behind this is that interfaces defined in pkg/actors/internal will be used to form a Building Block / Client which will implement all these methods defined in these interfaces.
+
+Whereas, methods defined for Services in protos will be implemented by respective new/existing Control Plane Service(s).
+
+And, the methods from these clients will call methods of the servers, wherever required.
+
+![actors_reminders](./resources/0011-R-multiple_reminders_actors_impl/Actors_Reminders_Interface.png)
+
+Depending on the implementation, it can be decided that what all RPCs need to be defined for a Service. Example, for a Placement Service which just creates and pushes Actor Types v/s Dapr Hosts table information to All sidecars, only table related ReportDaprStatusneeds to be as rpc.
```

Comment:
Can you clarify what is meant by `v/s`, is that meant as versus? Either way I am not sure I understand your example sentence. Can you reword this sentence to be clearer?

```suggestion
Depending on the implementation, it can be decided that all RPCs need to be defined for a Service. For example, for the Placement Service which creates and pushes Actor Types v/s Dapr Hosts table information to all sidecars, only table related ReportDaprStatus needs to be an rpc.
```